src/mixins/amalgamation-mixin.ts

```diff
@@ -41,6 +45,87 @@ export default class AmalgamationMixin extends Vue {
     )
   }
 
+  /**
+   * Get the business information, mailing address, email, and first filing if in LEAR.
+   * Otherwise, return error.
+   */
+  async fetchBusinessInfoForTing (item: any): Promise<any> {
+    // Get the auth info, business info, addresses and filings in parallel.
+    // Return data array; if any call failed, that item will be undefined.
+    const data = await Promise.allSettled([
+      AuthServices.fetchAuthInfo(item.identifier),
+      LegalServices.fetchBusinessInfo(item.identifier),
+      LegalServices.fetchAddresses(item.identifier),
+      LegalServices.fetchFirstOrOnlyFiling(item.identifier)
+    ]).then(results => results.map((result: any) => result.value))
+
+    const authInfo = data[0]
+    const businessInfo = data[1]
+    const addresses = data[2]
+    const firstFiling = data[3]
+
+    return {
+      authInfo,
+      businessInfo,
+      addresses,
+      firstFiling
+    }
+  }
+
+  /**
+   * If there is a state filing and restoration expiry date isn't in the past and the state filing is a
+   * limited restoration or limited restoration extension, then this business is in limited restoration.
+   * @param business The business to check if is in Limited Restoration or not.
+   */
+  isLimitedRestoration = async (business: any): Promise<boolean> => {
+    // check for no state filing
+    if (!business.businessInfo.stateFiling) return false
+    // check for expired restoration
+    if (this.getCurrentDate > business.businessInfo.restorationExpiryDate) return false
+    // fetch state filing
+    const stateFiling = await LegalServices.fetchFiling(business.businessInfo.stateFiling)
+    return (
+      stateFiling.restoration.type === RestorationTypes.LIMITED ||
+      stateFiling.restoration.type === RestorationTypes.LTD_EXTEND
+    )
+  }
+
+  /**
+   * Re-fetch the draft amalgamating businesses information and set in the store.
+   * Need to do that because the businesses might have changed from last draft save.
+   */
+  async refetchAmalgamatingBusinessesInfo (): Promise<void> {
+    const fetchTingInfo = async (item: any): Promise<AmalgamatingBusinessIF> => {
```

Comment:
If you don't love this style of function declaration, your could try a nested "function x (): ...".

Reply:
Yup, that's the one I usually use but I also like your way too! I grabbed this style from your isLimitedRestoration function. Trying to something new 😎